server/server.go

```diff
@@ -230,7 +206,19 @@ func (s *Server) indexGet(w http.ResponseWriter, r *http.Request) {
 	if len(qStrs) > 0 {
 		qStr := qStrs[len(qStrs)-1]
 		t0 := time.Now()
-		results, err := s.execQuery(qStr)
+
+		var results resultSetV1
+		txn, err := s.store.NewTransaction()
+
+		if err == nil {
+			var query ast.Body
+			_, query, err = ast.CompileQuery(qStr)
+			if err == nil {
+				results, err = s.execQuery(txn, query)
+			}
+			s.store.Close(txn)
+		}
```

Comment:
Yes, if the transaction fails, or the query doesn't compile, or the query execution fails, the error message is sent back in the results. This particular bit is for rendering the HTML you see if you point your browser at OPA (w/o any API path).